ui/src/main/java/edu/wpi/grip/ui/DeployController.java

```diff
@@ -0,0 +1,216 @@
+package edu.wpi.grip.ui;
+
+import com.google.common.eventbus.EventBus;
+import com.google.common.eventbus.Subscribe;
+import com.google.common.io.LineReader;
+import edu.wpi.grip.core.Pipeline;
+import edu.wpi.grip.core.events.ProjectSettingsChangedEvent;
+import edu.wpi.grip.core.events.StopPipelineEvent;
+import edu.wpi.grip.core.serialization.Project;
+import edu.wpi.grip.core.settings.ProjectSettings;
+import edu.wpi.grip.ui.util.StringInMemoryFile;
+import javafx.application.Platform;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.fxml.FXML;
+import javafx.scene.control.*;
+import net.schmizz.sshj.SSHClient;
+import net.schmizz.sshj.common.IOUtils;
+import net.schmizz.sshj.common.StreamCopier;
+import net.schmizz.sshj.connection.channel.direct.Session;
+import net.schmizz.sshj.userauth.UserAuthException;
+import net.schmizz.sshj.xfer.FileSystemFile;
+import net.schmizz.sshj.xfer.LoggingTransferListener;
+import net.schmizz.sshj.xfer.scp.SCPFileTransfer;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.InterruptedIOException;
+import java.io.StringWriter;
+import java.net.UnknownHostException;
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * JavaFX controller for the deploy tool.
+ * <p>
+ * This basically uploads a headless version of GRIP and the current project to a remote target using SSH, then
+ * runs it remotely.  The default values for all fields are based on typical settings for FRC, with the address
+ * based on the project settings.
+ */
+public class DeployController {
+    private final static String GRIP_JAR = "grip.jar";
+    private final static String PROJECT_FILE = "project.grip";
+
+    @FXML private TextField address;
+    @FXML private TextField user;
+    @FXML private TextField password;
+    @FXML private TextField javaHome;
+    @FXML private TextField deployDir;
+    @FXML private ProgressIndicator progress;
+    @FXML private Button deployButton;
+    @FXML private Label status;
+    @FXML private TextArea console;
+
+    @Inject private EventBus eventBus;
+    @Inject private Project project;
+    @Inject private Pipeline pipeline;
+    @Inject private Logger logger;
```

Comment:
> We haven't really made an official decision on styling

Like I said, this is true.

> We decided originally to use the default for Intellij.

Like both of us said, this isn't true.  Although I have recommended using IntelliJ's autoformatter, we never said our official style guide was the default settings in IntelliJ, and in practice our codebase more closely matches Google Java Style.

>If you want to change it then we need to talk about it as a team.

I'm not changing anything.  This code does not violate any official style guidelines we have, or any written but unofficial ones, and it's consistent with our current code.  If we want to establish an official style guide, I would think *that* would involve meeting as a team (although really I don't see why anyone would object to officially adopting Google's style)